sqlx-core/src/mysql/types/bigdecimal.rs

```diff
@@ -0,0 +1,193 @@
+        }
+    }
+
+    /// 15, -2 => 1500
+    /// 15, 1 => 1.5
+    /// 15, 2 => 0.15
+    /// 15, 3 => 0.015
+
+    fn size_hint(&self) -> usize {
+        let (bi, scale) = self.as_bigint_and_exponent();
+        let (_, radix) = bi.to_radix_be(10);
+        let mut s = radix.len();
+
+        if scale < 0 {
+            s = s + (-scale) as usize
+        } else if scale > 0 {
+            let scale = scale as usize;
+            if scale >= s {
+                s = scale + 1
+            }
+            s = s + 1;
+        }
+
+        if self.is_negative() {
+            s = s + 1;
+        }
+        s + 1
+    }
+}
+
+impl Decode<'_, MySql> for BigDecimal {
+    fn decode(value: MySqlValue) -> crate::Result<Self> {
+        match value.try_get()? {
+            MySqlData::Binary(mut binary) => {
+                let len = binary.get_u8()?;
+                let mut negative = false;
+                let mut scale: Option<i64> = None;
+                let mut v: Vec<u8> = Vec::with_capacity(len as usize);
+
+                loop {
+                    if binary.len() < 1 {
+                        break;
+                    }
+                    let data = binary.get_u8()?;
+                    match data {
+                        SIGN_NEG => {
+                            if !negative {
+                                negative = true;
+                            } else {
+                                return Err(Error::Decode(
+                                    format!("Unexpected byte: {:X?}", data).into(),
+                                ));
+                            }
+                        }
+                        SCALE_START => {
+                            if scale.is_none() {
+                                scale = Some(0);
+                            } else {
+                                return Err(Error::Decode(
+                                    format!("Unexpected byte: {:X?}", data).into(),
+                                ));
+                            }
+                        }
+                        0x30..=0x39 => {
+                            scale = scale.map(|s| s + 1);
+                            v.push(data - 0x30);
+                        }
+                        _ => {
+                            return Err(Error::Decode(
+                                format!("Unexpected byte: {:X?}", data).into(),
+                            ))
+                        }
+                    }
+                }
+
+                let r = BigInt::from_radix_be(
+                    if negative { Sign::Minus } else { Sign::Plus },
+                    v.as_slice(),
+                    10,
+                )
+                .ok_or(Error::Decode("Can't convert to BigInt".into()))?;
+
+                Ok(BigDecimal::new(r, scale.unwrap_or(0)))
+            }
+            MySqlData::Text(_) => Err(Error::Decode(
+                "`BigDecimal` can only be decoded from the binary protocol".into(),
```

Comment:
We'll want to figure this out unless this is a MySQL limitation. It should probably just be like `BigDecimal::from_str` I think?

Reply:
Text format is supported now, and I simplify the Encode/Decode implementation.